app/cosid/main_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestCompileAndRun(t *testing.T) {
```

Comment:
Test doesn't clean up correctly. Compiled binary gets deleted but the process (`cosid`) is still running after the test finishes.